### DISCUSSION TITLE
Strong name errors on initial clone and build

### DISCUSSION THREAD

**Discussion by sandcastle** [SixLabors/ImageSharp.Web#113] [Q&A] [answered]
@JimBobSquarePants - I have just cloned the repo to add GCP provider support and on the initial build of the solution after initialising the git modules I'm getting a strong name error. I'm not sure if this should be the default experience when building the solution, should it just be a CI build time consideration? It makes it a little difficult to get up and running and contribute a PR if you cant build the solution straight away.

Let me know how I can help improve the experience.

```
/Users/glenn/dev/ImageSharp.Web/artifacts/obj/src/ImageSharp.Web/Debug/netcoreapp3.1/ImageSharp.Web.AssemblyInfo.cs(25,12): error CS1726: Friend assembly reference 'SixLabors.ImageSharp.Web.Tests' is invalid. Strong-name signed assemblies must specify a public key in their InternalsVisibleTo declarations. [/Users/glenn/dev/ImageSharp.Web/src/ImageSharp.Web/ImageSharp.Web.csproj]
/Users/glenn/dev/ImageSharp.Web/artifacts/obj/src/ImageSharp.Web/Debug/netcoreapp3.1/ImageSharp.Web.AssemblyInfo.cs(26,12): error CS1726: Friend assembly reference 'ImageSharp.Web.Benchmarks' is invalid. Strong-name signed assemblies must specify a public key in their InternalsVisibleTo declarations. [/Users/glenn/dev/ImageSharp.Web/src/ImageSharp.Web/ImageSharp.Web.csproj]
/Users/glenn/dev/ImageSharp.Web/artifacts/obj/src/ImageSharp.Web/Debug/netcoreapp3.1/ImageSharp.Web.AssemblyInfo.cs(24,12): error CS1726: Friend assembly reference 'DynamicProxyGenAssembly2' is invalid. Strong-name signed assemblies must specify a public key in their InternalsVisibleTo declarations. [/Users/glenn/dev/ImageSharp.Web/src/ImageSharp.Web/ImageSharp.Web.csproj]
/Users/glenn/dev/ImageSharp.Web/artifacts/obj/src/ImageSharp.Web/Debug/netcoreapp2.1/ImageSharp.Web.AssemblyInfo.cs(25,12): error CS1726: Friend assembly reference 'SixLabors.ImageSharp.Web.Tests' is invalid. Strong-name signed assemblies must specify a public key in their InternalsVisibleTo declarations. [/Users/glenn/dev/ImageSharp.Web/src/ImageSharp.Web/ImageSharp.Web.csproj]
/Users/glenn/dev/ImageSharp.Web/artifacts/obj/src/ImageSharp.Web/Debug/netcoreapp2.1/ImageSharp.Web.AssemblyInfo.cs(26,12): error CS1726: Friend assembly reference 'ImageSharp.Web.Benchmarks' is invalid. Strong-name signed assemblies must specify a public key in their InternalsVisibleTo declarations. [/Users/glenn/dev/ImageSharp.Web/src/ImageSharp.Web/ImageSharp.Web.csproj]
/Users/glenn/dev/ImageSharp.Web/artifacts/obj/src/ImageSharp.Web/Debug/netcoreapp2.1/ImageSharp.Web.AssemblyInfo.cs(24,12): error CS1726: Friend assembly reference 'DynamicProxyGenAssembly2' is invalid. Strong-name signed assemblies must specify a public key in their InternalsVisibleTo declarations. [/Users/glenn/dev/ImageSharp.Web/src/ImageSharp.Web/ImageSharp.Web.csproj]
    0 Warning(s)
    6 Error(s)
```

**Comment by JimBobSquarePants** [comment, marked as answer]
Hi @sandcastle 

You should read the [Manual Build](https://github.com/SixLabors/ImageSharp/blob/9b9e13436da942e59c27d1f90ecf5632686e24c3/README.md#manual-build) section of the README. You haven't initialized the git submodules. They certainly complicate things but I don't know of a better way to share the code we need to across multiple repositories.

**Reply by sandcastle** [reply]
@JimBobSquarePants - not sure what I'm missing - I had already initialised the git modules as per read me, tried again and same issue. 

See below (db is just an alias for `dotnet build`):

<img width="1150" alt="image" src="https://user-images.githubusercontent.com/668657/91819467-6e546500-ec68-11ea-90d8-66763bd14f2c.png">

Current setup:

<img width="1158" alt="image" src="https://user-images.githubusercontent.com/668657/91820559-bd01ff00-ec68-11ea-896a-530a7aa0cb15.png">

**Reply by JimBobSquarePants** [reply]
Hold on, this is ImageSharp.Web! I'm transferring this discussion to the correct repository.

**Reply by JimBobSquarePants** [reply]
It's your .Net Core SDK preview. We should add a global.json with `allowprerelease` disabled. Hopefully we don't have to declare a specific SDK version (I don't want to do that)

https://docs.microsoft.com/en-us/dotnet/core/tools/global-json?tabs=netcore3x#allowprerelease

Temporarily you can do this.

![](https://user-images.githubusercontent.com/10199417/90156737-1921e380-dd8d-11ea-8fbc-0075d6de8ae9.png)

**Reply by sandcastle** [reply]
Great - that did the trick!

```json
{
  "sdk": {
    "allowPrerelease": false
  }
}
```

**Reply by JimBobSquarePants** [reply]
Ace, i'll use that elsewhere then!

I'm going to see if I can find out why the NET5 preview does this. We're definitely specifying a public key.

![image](https://user-images.githubusercontent.com/385879/91823639-71802e80-ec31-11ea-967c-b736c48e890d.png)

**Reply by JimBobSquarePants** [reply]
Issue raised here. https://github.com/dotnet/sdk/issues/13289
